packages/server/lib/browsers/utils.ts

```diff
@@ -338,12 +338,44 @@ const parseBrowserOption = (opt) => {
   }
 }
 
+/**
+ * Cast a wide net when checking for browser support - this should avoid all false negatives
+ * The regex below will match: msedge.exe, edge, edge-beta
+ */
+function createSupportedBrowserRegex (browsers: string[]): RegExp {
```

Comment:
I'm remembering that @emilyrohrbough had a draft PR to fix the supported browsers list because it's currently not accurate in what it says we support. https://github.com/cypress-io/cypress/pull/28377/files Maybe she can comment or this code will be helpful somehow.

Reply:
That was very helpful thank you - a few thoughts around adding `webkit` and `electron` directly to the `knownBrowsers` object:
- Does the Electron browser have a binary or a versionRegex? If not, I wonder if it might be better to export it separately in `known-browsers` with its own type defined in `types/src/browsers.ts` e.g. `export type ElectronBrowser = Omit<Browser, 'versionRegex' | 'binary'>`. We could make the `binary` and `versionRegex` properties in `Browser` optional, but that would require a few changes in `launcher/lib/detect.ts` to avoid runtime errors (not sure what else we would have to change).

- The Webkit browser in @emilyrohrbough draft PR conforms to the `Browser` type, but adding it directly to the `knownBrowsers` object will require some changes to `launcher/lib/windows/index.ts` and `launcher/lib/darwin/index.ts` in order to get the `darwin_spec.ts` and `windows_spec.ts` tests passing successfully.

- Adding these both directly to the `knownBrowsers` I believe will require some changes to browser detection in `server/lib/browsers/utils.ts`, specifically in `getBrowsers()`

Let me know if you're happy to keep both webkit and electron in `known-browsers` but exported separately as a temporary solution, or if there is a better solution that I'm missing entirely (I could just be overthinking this).